ed25519-donna/ed25519.c

```diff
@@ -62,6 +62,7 @@ ED25519_FN(ed25519_sign) (const unsigned char *m, size_t mlen, const ed25519_sec
 	ge25519 ALIGN(16) R;
 	hash_512bits extsk, hashr, hram;
 	unsigned char randr[32];
+	static const unsigned char rzero[64] = {0};
```

Comment:
Are these attacker unknown bytes?  This looks like 64 zeros.

Reply:
Yeah it's just zeroes. The paper recommends padding it with zeroes or another constant to reach the 128 byte block size. I think the principle is that the next round will start with attacker unknown values.

Reply:
Should we assert mlen is 32 to ensure that padding is correct, and also include a comment to this effect
```suggestion
	/*
	 * Pad the message to 128 bytes in size; 32-byte signature; 32-byte random; 64-byte zero
	 */
	static const unsigned char rzero[64] = {0};
	assert (mlen == 32);
```

Reply:
mlen is irrelevant here because that comes after the anti-attacker block